### DISCUSSION TITLE
code auto formatting

### DISCUSSION THREAD

**Discussion by amigo421** [microsoft/vscode-cpptools#11217] [Q&A]
not sure that question is cpp extension responsibility

I'm reading [this](https://code.visualstudio.com/docs/cpp/cpp-ide#_code-formatting)
there is such triggers:
- editor.formatOnSave - to format when you save your file.
- editor.formatOnType - to format as you type (triggered on the ; character).

however in other IDEs is more native trigger (for CPP)  - closing bracket "}" 
this symbol  usually  triggers to re-format whole code block above

is it possible to set up this symbol as a trigger for activate a formatting ?

**Comment by sean-mcmanus** [comment]
Yes, it's possible to auto-format on }, but our clang-tidy formatting mode doesn't support that -- it looks like our vcFormat mode may support that, but I haven't tried it out. I don't remember exactly why } wasn't not supported for clang-format, but we have a feature request possibly adding that support at https://github.com/microsoft/vscode-cpptools/issues/1419 .
